lib/github_changelog_generator/gitlab_fetcher.rb

```diff
@@ -0,0 +1,449 @@
+      end
+      print_empty_line
+
+      if tags.count == 0
+        Helper.log.warn "Warning: Can't find any tags in repo. \
+Make sure, that you push tags to remote repo via 'git push --tags'"
+      else
+        Helper.log.info "Found #{tags.count} tags"
+      end
+      # tags are a Sawyer::Resource. Convert to hash
+      tags.map { |resource| stringify_keys_deep(resource.to_hash) }
+    end
+
+    def closed_pr_options
+      @closed_pr_options ||= {
+        filter: "all", labels: nil, state: "merged"
+      }.tap { |options| options[:since] = @since if @since }
+    end
+
+    # This method fetch all closed issues and separate them to pull requests and pure issues
+    # (pull request is kind of issue in term of GitHub)
+    #
+    # @return [Tuple] with (issues [Array <Hash>], pull-requests [Array <Hash>])
+    def fetch_closed_issues_and_pr
+      issues = []
+      print "Fetching closed issues...\r" if @options[:verbose]
+      options = { state: "closed", scope: :all }
+      @client.issues(@project_id, DEFAULT_REQUEST_OPTIONS.merge(options)).auto_paginate do |issue|
+        issue = stringify_keys_deep(issue.to_hash)
+        issue["body"] = issue["description"]
+        issue["html_url"] = issue["web_url"]
+        issue["number"] = issue["iid"]
+        issues.push(issue)
+      end
+
+      print_empty_line
+      Helper.log.info "Received issues: #{issues.count}"
+
+      # separate arrays of issues and pull requests:
+      [issues.map { |issue| stringify_keys_deep(issue.to_hash) }, fetch_closed_pull_requests]
+    end
+
+    # Fetch all pull requests. We need them to detect :merged_at parameter
+    #
+    # @return [Array <Hash>] all pull requests
+    def fetch_closed_pull_requests
+      pull_requests = []
+      options = { state: "merged", scope: :all }
+
+      @client.merge_requests(@project_id, options).auto_paginate do |new_pr|
+        new_pr = stringify_keys_deep(new_pr.to_hash)
+        # align with Github naming
+        new_pr["number"] = new_pr["iid"]
+        new_pr["html_url"] = new_pr["web_url"]
+        new_pr["merged_at"] = new_pr["updated_at"]
+        new_pr["pull_request"] = true
+        new_pr["user"] = { login: new_pr["author"]["username"], html_url: new_pr["author"]["web_url"] }
+        # to make it work with older gitlab version or repos that lived across versions
+        new_pr["merge_commit_sha"] = new_pr["merge_commit_sha"].nil? ? new_pr["sha"]: new_pr["merge_commit_sha"]
+        pull_requests.push(new_pr)
+      end
+
+      print_empty_line
+
+      Helper.log.info "Pull Request count: #{pull_requests.count}"
+      pull_requests.map { |pull_request| stringify_keys_deep(pull_request.to_hash) }
+    end
+
+    # Fetch event for all issues and add them to 'events'
+    #
+    # @param [Array] issues
+    # @return [Void]
+    def fetch_events_async(issues)
+      i       = 0
+      threads = []
+      options = { }
+      if issues.empty?
+        return
+      end
```

Comment:
Style: Make this guard clause on one line.

`return if issues.empty?`

Follow it by a blank line.

Reply:
Yeah, RuboCop flagged this one up as well.